File: XamarinStore.iOS/Helpers/UIViewExtensions.cs
using System;
using MonoTouch.UIKit;

namespace XamarinStore
{
	public static class UIViewExtensions
	{
		public static UIViewController GetParentViewController(this UIView view)
		{
			var nextResponder = view.NextResponder;
			if (nextResponder is UIViewController)
				return (UIViewController)nextResponder;
			else if (nextResponder is UIView)
				return GetParentViewController ((UIView)nextResponder);
			return null;
		}
	}
}

File: F#/XamarinStore.iOS/Helpers/UIViewExtensions.fs
namespace XamarinStore

open System
open MonoTouch.UIKit

module UIViewHelpers =

    let rec private GetParentViewController (view:UIView) =
            match view.NextResponder with
            | :? UIViewController as nextResponder -> nextResponder
            | :? UIView as view ->GetParentViewController view
            | _ -> null

    type UIView with
        member this.GetParentViewController() =
            GetParentViewController this